editor/test/integration/game_object_test.clj

```diff
@@ -60,18 +61,24 @@
 (deftest embedded-components
   (test-util/with-loaded-project
     (let [resource-types (game-object/embeddable-component-resource-types workspace)
-          save-data (partial save-data project)
-          make-restore-point! #(test-util/make-graph-reverter (project/graph project))
-          add-component! (partial test-util/add-embedded-component! app-view nil)
+          add-component! (partial test-util/add-embedded-component! app-view)
           go-id (project/get-resource-node project "/game_object/test.go")
-          go-resource (g/node-value go-id :resource)]
+          go-resource (g/node-value go-id :resource)
+          go-read-fn (:read-fn (resource/resource-type go-resource))]
       (doseq [resource-type resource-types]
         (testing (:label resource-type)
-          (with-open [_ (make-restore-point!)]
+          (with-open [_ (test-util/make-graph-reverter (project/graph project))]
             (add-component! resource-type go-id)
-            (let [saved-string (:content (save-data go-resource))
-                  saved-embedded-components (g/node-value go-id :embed-ddf)
-                  loaded-embedded-components (:embedded-components (protobuf/read-text GameObject$PrototypeDesc (StringReader. saved-string)))
+            (let [save-value (g/node-value go-id :save-value)
+                  load-value (with-open [reader (StringReader. (:content (g/node-value go-id :save-data)))]
+                               (go-read-fn reader))
+                  saved-embedded-components (:embedded-components save-value)
+                  loaded-embedded-components (:embedded-components load-value)
                   [only-in-saved only-in-loaded] (data/diff saved-embedded-components loaded-embedded-components)]
               (is (nil? only-in-saved))
-              (is (nil? only-in-loaded)))))))))
+              (is (nil? only-in-loaded))
+              (when (or (some? only-in-saved)
+                        (some? only-in-loaded))
+                (println "When comparing" (:label resource-type))
+                (prn 'disk only-in-loaded)
+                (prn 'save only-in-saved)))))))))
```

Comment:
This test was testing the wrong thing by being too low-level. Rewritten to make use of the full `read-fn` and `save-data` output of the `GameObjectNode`.